src/components/Cart/Add-To-Cart.jsx

```diff
@@ -0,0 +1,48 @@
+import { useAuthContext, useData } from "../../context/index";
+import { useNavigate } from "react-router-dom";
+import { updateWishlist, updateCart } from "./../../services/index";
+import { itemExists } from "../../Utils/arrays-utility";
+
+const itemExistsInCart = (cartItems, productId) =>
+    cartItems.some(
+        (cartItem) => cartItem._id === productId && cartItem.quantity > 0
+    );
+
+export const AddToCart = ({ product, btnNormalCss, btnSuccessCss }) => {
+
+    const {
+        state: { itemsInCart, itemsInWishlist },
+        dispatch, setIsError
+    } = useData();
+    const { login, setShowLoader } = useAuthContext();
+    const isItemInCart = itemExistsInCart(itemsInCart, product._id);
+    const navigate = useNavigate();
+    const isWishlisted = itemExists(itemsInWishlist, product._id);
+    const updateLists = () => {
+        updateWishlist(product, isWishlisted, dispatch, setShowLoader, setIsError);
+        updateCart(product, "ADD", dispatch, setShowLoader, setIsError);
+    };
+    return (
+        <>
+            <button
+                type="button"
+                disabled={!product.inStock}
+                className={isItemInCart ? btnNormalCss : btnSuccessCss}
```

Comment:
one suggestion will be the styles for `btnNormalCss` and `btnSuccess` CSS can be specified here right Their no need to send in props.

Reply:
Hey here, I'm changing the button css class name based on itemsInCart & I'm using AddToCart Multiple times that's why I'm sending through props